js/src/data/resolvers.js

```diff
@@ -148,10 +148,37 @@ export function* getTargetAudience() {
 	yield fetchTargetAudience();
 }
 
-export function* getAdsCampaigns() {
-	yield fetchAdsCampaigns();
+export function* getAdsCampaigns( query ) {
+	try {
+		const campaigns = yield apiFetch( {
+			path: addQueryArgs( `${ API_NAMESPACE }/ads/campaigns`, query ),
+		} );
+
+		return {
+			type: TYPES.RECEIVE_ADS_CAMPAIGNS,
+			query,
+			adsCampaigns: campaigns.map( adaptAdsCampaign ),
+		};
+	} catch ( error ) {
+		yield handleFetchError(
+			error,
+			__(
+				'There was an error loading ads campaigns.',
+				'google-listings-and-ads'
+			)
+		);
+	}
 }
 
+getAdsCampaigns.shouldInvalidate = ( action, query ) => {
+	return (
+		( action.type === TYPES.UPDATE_ADS_CAMPAIGN ||
+			action.type === TYPES.DELETE_ADS_CAMPAIGN ||
+			action.type === TYPES.CREATE_ADS_CAMPAIGN ) &&
+		query?.exclude_removed === false
+	);
+};
```

Comment:
❓  Why is this for?

Reply:
In case the ads campaign is updated, deleted or created, we will re-fetch the data for `all_ads_campaigns`. For example, if you ad a new Ads campaign so it will show the new ads campaign in the filter section.